src/win/util.c

```diff
@@ -1195,27 +1195,44 @@ int uv_os_homedir(char* buffer, size_t* size) {
     return 0;
   }
 
-  /* USERPROFILE is not set, so call SHGetKnownFolderPath() */
-  if (SHGetKnownFolderPath(&FOLDERID_Profile, 0, NULL, &path) != S_OK)
-    return uv_translate_sys_error(GetLastError());
+  /* USERPROFILE is not set, so call GetUserProfileDirectoryW() */
+  if (OpenProcessToken(GetCurrentProcess(), TOKEN_READ, &hToken) == 0) {
+    r = GetLastError();
+    free(path);
+    return uv_translate_sys_error(r);
+  }
+
+  if (!GetUserProfileDirectoryW(hToken, path, size)) {
+    r = GetLastError();
+    free(path);
+    CloseHandle(hToken);
+
+    if (r == ERROR_INSUFFICIENT_BUFFER) {
+      *size = *size - 1;
+      return UV_ENOBUFS;
+    }
 
+    return uv_translate_sys_error(r);
+  }
+
+  CloseHandle(hToken);
   bufsize = uv_utf16_to_utf8(path, -1, buffer, *size);
 
   if (bufsize == 0) {
     r = GetLastError();
+    free(path);
 
     if (r == ERROR_INSUFFICIENT_BUFFER) {
-      *size = wcslen(path);
-      CoTaskMemFree(path);
+      *size = *size - 1;
       return UV_ENOBUFS;
     }
 
-    CoTaskMemFree(path);
     return uv_translate_sys_error(r);
   }
 
-  CoTaskMemFree(path);
-  *size = bufsize - 1;
+  free(path);
+  assert(*size == bufsize);
+  *size = *size - 1;
```

Comment:
I think these 2 last statements are wrong. The previous statement was correct: `*size = bufsize - 1`, because `uv_utf16_to_utf8` won't write to `size`, but return how much it has written.